ocaml/idl/datamodel_observer.ml

```diff
@@ -0,0 +1,157 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let register =
+  call ~name:"register" ~in_oss_since:None ~lifecycle:[] ~hide_from_docs:true
+    ~doc:"Register a new observer on all hosts"
+    ~params:
+      [
+        (Ref _observer, "self", "The observer")
+      ; (Ref _host, "host", "The host to be registered on")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let unregister =
+  call ~name:"unregister" ~in_oss_since:None ~lifecycle:[] ~hide_from_docs:true
+    ~doc:"Unegister a new observer on all hosts"
+    ~params:
+      [
+        (Ref _observer, "self", "The observer")
+      ; (Ref _host, "host", "The host to be registered on")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_hosts =
+  call ~name:"set_hosts" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Sets the hosts that the observer is to be registered on"
+    ~params:
+      [
+        (Ref _observer, "self", "The observer")
+      ; (Set (Ref _host), "value", "Hosts the observer is to be registered on")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_enabled =
+  call ~name:"set_enabled" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Enable / disable this observer which will stop the proivder from \
```

Comment:
```suggestion
      "Enable / disable this observer which will stop the provider from \
```

Reply:
should be observer, typo meant it wasn't changed

Reply:
Fixed